osimpipeline/task.py

```diff
@@ -96,29 +105,183 @@ def __init__(self, trial):
+            self.generate_setup_file()
+
+    def add_tool_dir(self):
+        if not os.path.exists(self.tool_path): os.makedirs(self.tool_path)
+
+    def add_cycle_dirs(self, cycles):
+        self.add_tool_dir()
+        for cycle in cycles:
+            cycle_path = os.path.join(self.tool_path, cycle.name)
+            if not os.path.exists(cycle_path): os.makedirs(cycle_path)
+
+    def fill_setup_template(self):
+        raise NotImplementedError()
+
+    def fill_external_loads_template(self, file_dep, target):
+        with open(file_dep[0]) as ft:
+            content = ft.read()
+            content = content.replace('@STUDYNAME@', self.study.name)
+            content = content.replace('@NAME@', self.trial.id)
+
+        with open(target[0], 'w') as f:
+            f.write(content)
+
+    def copy_over_external_loads(self):
+        if not os.path.exists(self.source_extloads_fpath):
+            # The user does not yet have a external_loads.xml in place; fill 
+            # out the template.
+            self.add_action(
+                ['templates/%s/external_loads.xml' % self.tool],
+                [self.source_extloads_fpath],
+                self.fill_external_loads_template)
+            self.actions.append((self.copy_file,
+                [[self.source_extloads_fpath], [self.results_extloads_fpath]]))
+        else:
+            # We have already filled out the template tasks file,
+            # and the user might have made changes to it.
+            self.add_action(
+                    [self.source_extloads_fpath],
+                    [self.results_extloads_fpath],
+                    self.copy_file)
+
+    def fill_tasks_template(self, file_dep, target):
+        # Don't overwrite existing tasks file
+        if not os.path.exists(target[0]):
+            with open(file_dep[0]) as ft:
+                content = ft.read()
+                content = content.replace('@STUDYNAME@', self.study.name)
+                content = content.replace('@NAME@', self.trial.id)
+    
+            with open(target[0], 'w') as f:
+                f.write(content)
+
+    def copy_over_tasks(self):
+        if not os.path.exists(self.source_tasks_fpath):
+            # The user does not yet have a tasks.xml in place; fill out the
+            # template.
+            self.add_action(
+                    ['templates/%s/tasks.xml' % self.tool],
+                    [self.source_tasks_fpath],
+                    self.fill_tasks_template)
+            self.actions.append((self.copy_file,
+                [[self.source_tasks_fpath], [self.results_tasks_fpath]]))
+        else:
+            # We have already filled out the template tasks file,
+            # and the user might have made changes to it.
+            self.add_action(
+                    [self.source_tasks_fpath],
+                    [self.results_tasks_fpath],
+                    self.copy_file)
+
+    def generate_setup_file(self, cycle=None):
```

Comment:
Could you rename this to indicate that it's creating an action rather than doing the action itself? Perhaps `create_setup_file_action`.